### PR TITLE
Promotions: Add export helper functions

### PR DESCRIPTION
**Pull request by coderkevin**
This adds functions that convert promotion objects to their respective
product or coupon API update objects.

To Test: `npm test`

### CODE REVIEW

File: client/extensions/woocommerce/state/sites/promotions/helpers.js [hunk 1]
```diff
@@ -15,6 +15,22 @@ export function createPromotionFromProduct( product ) {
 	};
 }
 
+export function createProductUpdateFromPromotion( promotion ) {
+	const { productIds } = promotion.appliesTo;
```
Comment (timmyc) [inline client/extensions/woocommerce/state/sites/promotions/helpers.js]: Should we check for `promotion` here or maybe default the argument to `{}`?

File: client/extensions/woocommerce/state/sites/promotions/helpers.js [hunk 2]
```diff
@@ -45,6 +61,32 @@ export function createPromotionFromCoupon( coupon ) {
 	return promotion;
 }
 
+export function createCouponUpdateFromPromotion( promotion ) {
+	if ( ! promotion.couponCode ) {
```
Comment (timmyc) [inline client/extensions/woocommerce/state/sites/promotions/helpers.js]: Same here, maybe also add `! promotion ` to this clause?

File: client/extensions/woocommerce/state/sites/promotions/helpers.js (hunk 1 above)
Comment (coderkevin) [reply client/extensions/woocommerce/state/sites/promotions/helpers.js]: This is in the category of "should never happen", but sure, I'll add it.

File: client/extensions/woocommerce/state/sites/promotions/helpers.js (hunk 2 above)
Comment (coderkevin) [reply client/extensions/woocommerce/state/sites/promotions/helpers.js]: I think if there is code passing in null/undefined as a promotion here, the proper behavior is to crash, honestly.

File: client/extensions/woocommerce/state/sites/promotions/helpers.js (hunk 1 above)
Comment (timmyc) [reply client/extensions/woocommerce/state/sites/promotions/helpers.js]: Ha, kind of figured as much - feel free to disregard then.

**Comment by coderkevin** [comment]
Rebased and added extra protection for promotions without a `appliesTo` property.

**Pull request merged**